custom_components/bosch_alarm/sensor.py

```diff
@@ -0,0 +1,64 @@
+""" Support for Bosch Alarm Panel History as a sensor """
+
+from __future__ import annotations
+
+import logging
+
+from homeassistant.components.sensor import (
+    SensorEntity
+)
+
+from homeassistant.const import EntityCategory
+
+from .const import (
+    DOMAIN,
+)
+from .device import device_info_from_panel
+
+_LOGGER = logging.getLogger(__name__)
+
+HISTORY_ATTR = 'history'
+HISTORY_ID_ATTR = 'history_id'
+
+class PanelHistorySensor(SensorEntity):
+    def __init__(self, panel, unique_id):
+        self._panel = panel
+        self._observer = panel.history_observer
+        self._unique_id = unique_id
+        self._attr_device_info = device_info_from_panel(panel)
+        self._attr_entity_category = EntityCategory.DIAGNOSTIC
+
+    @property
+    def icon(self): return "mdi:history"
+
+    @property
+    def unique_id(self): return self._unique_id
+
+    @property
+    def should_poll(self): return False
+
+    @property
+    def state(self): return len(self._panel.history)
+
+    @property
+    def name(self): return f"{self._panel.model} History"
+
+    @property
+    def extra_state_attributes(self):
+        history = self._panel.history
+        return { HISTORY_ATTR: "\n".join(history)}
```

Comment:
Let's mention how this works in the README, otherwise it's not very discoverable